src/drivers/driver_zephyr.c

```diff
@@ -1525,10 +1525,14 @@ static int wpa_drv_zep_set_supp_port(void *priv,
 				     authorized,
 				     if_ctx->bssid);
 
-#ifdef CONFIG_NET_DHCPV4
-	if (authorized) {
-		net_dhcpv4_restart(iface);
-    }
+#if defined(CONFIG_NET_DHCPV4) && defined(CONFIG_SME)
```

Comment:
SME is mainly for separate auth and assoc compared to a single connect. And also as I said in https://github.com/nrfconnect/sdk-nrf/pull/14718#discussion_r1617261406 this needs to be per-iface setting which is a bit complicated.

Thinking on this, this should really be application's responsibility, I added this in WPA supp as a hack to be common to all sample. WPA supplicant shouldn't be managing DHCP at all. May be just remove?

@rlubos the main reason for doing this was DHCP starts right after iface up, and by the time Wi-Fi is connected, it has hit 1min random backoff, so, any attempt over Wi-Fi reconnection takes about 1min.

DHCP client should ideally register for NET_IF events and always (re)start on UP_RUNNING, WDYT?

Reply:
I would recommend to add `CONFIG_NET_DHCPV4=n` in SAP configuration for time being and we can discuss on a proper solution that is part of the networking stack/application.

Reply:
> DHCP client should ideally register for NET_IF events and always (re)start on UP_RUNNING, WDYT?

It actually does that already (https://github.com/zephyrproject-rtos/zephyr/blob/53ddff639562ef68dc0a6f62b82f7505c75ebdce/subsys/net/lib/dhcpv4/dhcpv4.c#L1417), however the dhcpv4 have to be started already on the interface anyway.

Reply:
Thanks @rlubos , should we add a Kconfig option to do this on boot for the default iface, once it's UP_RUNNING? This should server for most usecases.

@rado17 We should test and explore this by removing the DHCP restart from WPA supplicant?

Reply:
Couldn't that be achieved by enabling`NET_CONFIG_SETTINGS`, but w/o any other configurations (I. e. no static address etc.). It should then start the DHCP client only. @jukkar thoughts?

Reply:
> Couldn't that be achieved by enabling`NET_CONFIG_SETTINGS`, but w/o any other configurations (I. e. no static address etc.). It should then start the DHCP client only

Yes, that is one possibility, but note that config lib is optional and if it is not enabled, application needs to start dhcp client manually.

Reply:
As removing this from WPA supplicant is invasive, as samples today rely on WPA supplicant to manage DHCP client, let's go ahead with runtime SME check and then in next release we can remove this and let samples manage DHCP.